### PR TITLE
Set the default IMU calibration to the SAE standard

### PR DESCRIPTION
**Pull request by stieg**
This change sets the default IMU config to conform to the SAE standard.
It also creates enum types for the channels and modes and ups the precision
on the gyro portions since this is now more sensitive.  It also addresses
the fallout from making these default changes.

Issues: #360

### CODE REVIEW

File: include/logger/loggerConfig.h [hunk 1]
```diff
@@ -216,59 +216,62 @@ typedef struct _GPIOConfig {
...
-         MODE_IMU_NORMAL,                       \
-         IMU_CHANNEL_YAW,                       \
-         DEFAULT_GYRO_ZERO,                     \
-         0.1F                                   \
-         }
+#define MIN_IMU_RAW	0
+#define MAX_IMU_RAW	4097
+
+#define DEFAULT_ACCEL_ZERO	2048
+/* LY330ALH zero state voltage output is 1.5v */
```
Comment (brentpicasso) [inline include/logger/loggerConfig.h]: LY330ALH is from MK1; I think we need to reset this number to be the actual yaw zero value for the MPU-9150 / 9250

File: include/logger/loggerConfig.h [hunk 2]
```diff
@@ -216,59 +216,62 @@ typedef struct _GPIOConfig {
...
+#define IMU_GYRO_CONFIG(name, mode, chan) {     \
+                IMU_GYRO_CH_CONFIG(name),       \
+                        mode,                   \
+                        chan,                   \
+                        DEFAULT_GYRO_ZERO,      \
+                        0.1F                    \
+                        }
 
+#define IMU_CONFIG_DEFAULTS {                                                 \
+                IMU_ACCEL_CONFIG("AccelX", IMU_MODE_INVERTED, IMU_CHANNEL_Y), \
```
Comment (brentpicasso) [inline include/logger/loggerConfig.h]: need some code comments here to explain why we're doing this mapping

**Comment by brentpicasso** [comment]
This PR should be re-worked to not do the SAE mappings at the higher level, but should be remapped within the HAL layer (imu_device_stm32.c), function at line 72  (imu_device_read). 

Rationale: If defined at a higher level, then users will see in the app config, some channels inverted or not inverted, which will appear confusing, and adding cognitive load to understanding that we're mapping ISO to SAE from the interface we're presenting them.

By doing it at a lower level we're in effect converting the IMU device to be "SAE compliant" . The result is, the default configuration for the IMU would have X mapped to X, Y mapped to Y, no channel inversions, etc.

File: include/logger/loggerConfig.h (hunk 2 above)
Comment (brentpicasso) [inline include/logger/loggerConfig.h]: NVM, see top level PR comment.

**Comment by stieg** [comment]
You are right.  Will re-tool the patch.

File: include/logger/loggerConfig.h (hunk 1 above)
Comment (stieg) [inline include/logger/loggerConfig.h]: Yeah something doesn't seem right for me here either.  I'll play with this a bit.

Comment (brentpicasso) [inline include/logger/loggerConfig.h]: A simple update to the zero value that's appropriate for the MPU-9150/MPU-9250 would do the trick I think.

Comment (stieg) [inline include/logger/loggerConfig.h]: It does.  In next patch.

**Comment by stieg** [comment]
@brentpicasso There we go.  Check it out now.

**Comment by brentpicasso** [comment]
Looks good. I assume it's been bench tested; go ahead and :shipit:

**Comment by stieg** [comment]
Yep.  Tested and verified.

On Thu, Feb 11, 2016 at 10:07 AM, Brent Picasso <notifications@github.com>
wrote:

> Looks good. I assume it's been bench tested; go ahead and [image:
> :shipit:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/385#issuecomment-182936154>
> .
>



--

**Pull request merged**